ros_ws/src/autonomous/src/emergency_stop.cpp

```diff
@@ -2,67 +2,53 @@
 
 EmergencyStop::EmergencyStop()
 {
-    lidar_subscriber =
+    m_lidar_subscriber =
         m_node_handle.subscribe<sensor_msgs::LaserScan>(TOPIC_LASER_SCAN, 1, &EmergencyStop::lidarCallback, this);
-    emer_stop_publisher = m_node_handle.advertise<std_msgs::Bool>(TOPIC_EMERGENCY_STOP, 1);
+    m_emergency_stop_publisher = m_node_handle.advertise<std_msgs::Bool>(TOPIC_EMERGENCY_STOP, 1);
 }
 
-float EmergencyStop::rangeAtDegree(const sensor_msgs::LaserScan::ConstPtr& lidar, float theta)
+bool EmergencyStop::emergencyStop(const sensor_msgs::LaserScan::ConstPtr& lidar)
 {
-    // check if calculated index is invalid
-    // if so then return the chosen max range (here 4)
-    if (theta / 0.375f < 0 || theta / 0.375f > 719)
-    {
-        return 4;
-    }
+    int available_samples = (lidar->angle_max - lidar->angle_min) / lidar->angle_increment;
+    int samples_used = 0;
+    float range_sum = 0;
 
-    // ranges size is 720, angle range is 270°
-    // so one step equals an angle of 0,375°
-    // to access range at angle theta, you have to divide the given theta by the step size
-    float range = lidar->ranges[theta / 0.375];
-
-    // check for absurd or nan values
-    // the max range should be 4 so the car doesn't get affected by walls very far away
-    if (range > 4 || std::isinf(range))
+    int index_start = available_samples / 2 - SAMPLE_ANGLE / lidar->angle_increment / 2;
+    int index_end = available_samples / 2 + SAMPLE_ANGLE / lidar->angle_increment / 2;
+    for (int i = index_start; i < index_end; i++)
```

Comment:
The old ```rangeAtDegree``` checked for invalid indexes. I think that was quite useful, but as the lidar does not changes on runtime it should not be a normal conditional and instead it could be realized via an assertion. This way the overhead of the additional condition is only present on debug builds.

Such a assertion could look like:
```C++
ROS_ASSERT_MSG(i >= 0 & i < static_cast<int>(lidar->angle_max / lidar->angle_increment), "Lidar data are accessed with invalid index. i = %d", i);
```